aten/src/ATen/core/function_schema.h

```diff
@@ -156,18 +156,22 @@ struct FunctionSchema {
     checkSchema();
   }
 
-  // check whether this schema is backward compatible with the old one.
-  // the following conditions are considered as this schema is backward
-  // compatible with old:
-  //   1) two schemas are equal
-  //   2) this schema has the same or more positional args than old,
-  //      and any positional arg in this schema is backward compatible
-  //      with the corresponding one in old schema, which could be an arg
-  //      or a kwarg, if it has, or it must provide a default value
-  //   3) this schema has the same or more kwargs than old, and all the kwargs
-  //      in old schema can find the corresponding kwarg in this schema which
-  //      is backward compatible with the old kwarg, and the extra kwargs in
-  //      this schema must provide default values.
+  // Checks whether this schema is backward compatible with the old one.
+  // The following conditions must be true:
+  // * The function's structure of old and new match (e.g. name,
```

Comment:
I think this could be a little clearer. A complete enumeration would be something like this, right?

- [Output Narrowing] The new schema's output type must be the same class or inherit from the old schema's output type
- [Arg Count] The new schema must have at least as many arguments as the old schema
- [Default Values] Every new argument must have a default value
- [ Arg Compatibility] Every argument in the old schema has a corresponding argument in the new schema that..
  - is at the same position
  - has the same name
  - has the same type, or the old argument's type inherits from the new argument's type

In the current description I think it's unclear what "matching" means.

Reply:
Thanks for the comment. I think what you suggest makes sense.

One thing though: I think simplifying the explanation just in terms of arguments as the list of all positional and kwargs makes it a 
bit simpler to think and avoid dealing with the individual concepts.
E.g. "The new schema must have exactly as many positional arguments as the old schema" is not quite the case since since a kwargs in old may have moved to be a positional in new (i.e. in the case f_new(a, b, *, c) => f_old(a, *, b, c) , f_new would have 2 positional vs f_old's 1). That's why there's not a direct requirement on number of positional arguments being the same.